src/core_plugins/kibana/public/home/components/sample_data_set_cards.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiFlexGrid,
+  EuiFlexItem,
+} from '@elastic/eui';
+
+import { SampleDataSetCard } from './sample_data_set_card';
+
+import {
+  listSampleDataSets,
+  installSampleDataSet,
+  uninstallSampleDataSet
+} from '../sample_data_sets';
+
+export class SampleDataSetCards extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sampleDataSets: [],
+    };
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async componentDidMount() {
+    this._isMounted = true;
+
+    this.loadSampleDataSets();
+  }
+
+  loadSampleDataSets = async () => {
+    const sampleDataSets = await listSampleDataSets();
+
+    if (!this._isMounted) {
+      return;
+    }
+
+    this.setState({
+      sampleDataSets: sampleDataSets
+        .map((sampleDataSet) => {
+          sampleDataSet.isProcessing = false;
```

Comment:
Looks like this is modifying the returned object.  When we have typescript, the structure of `sampleDataSet` will be easy to see, and at that point, I don't think it would make sense to put an `isProcessing` flag on it just for the purpose of this class.  What about a mapping that keeps track of `isProcessing` for each sample data set by id?